src/ctap2/mod.rs

```diff
@@ -809,5 +809,440 @@ pub(crate) fn set_or_change_pin_helper(
+                                );
+                        }
+                        unwrap_result!(cred_management.regenerate_puap(), callback);
+                        continue;
+                    }
+                    CredManagementCommand::EnumerateCredentialsBegin(..)
+                    | CredManagementCommand::EnumerateCredentialsGetNextCredential => {
+                        let user = unwrap_option!(result.user, callback);
+                        let credential_id = unwrap_option!(result.credential_id, callback);
+                        let public_key = unwrap_option!(result.public_key, callback);
+                        let cred_protect = unwrap_option!(result.cred_protect, callback);
+                        let large_blob_key = result.large_blob_key;
+
+                        if matches!(
+                            cred_management.subcommand,
+                            CredManagementCommand::EnumerateCredentialsBegin(..)
+                        ) {
+                            remaining_cred_ids =
+                                unwrap_option!(result.total_credentials, callback) - 1;
+                        } else {
+                            remaining_cred_ids -= 1;
+                        }
+                        // We might have to change the global variable, but need the unmodified below
+                        let current_rp_backup = current_rp;
+                        let mut we_are_done = false;
+                        if remaining_cred_ids > 0 {
+                            cred_management.subcommand =
+                                CredManagementCommand::EnumerateCredentialsGetNextCredential;
+                        } else {
+                            current_rp += 1;
+                            // We have all credentials from this RP. Starting with the next RP.
+                            if current_rp < credential_result.credential_list.len() {
+                                cred_management.subcommand =
+                                    CredManagementCommand::EnumerateCredentialsBegin(
+                                        credential_result.credential_list[current_rp]
+                                            .rp_id_hash
+                                            .clone(),
+                                    );
+                                unwrap_result!(cred_management.regenerate_puap(), callback);
+                            } else {
+                                // Finally done iterating over all RPs and their Credentials
+                                we_are_done = true;
+                            }
+                        }
+                        let key = CredentialListEntry {
+                            user,
+                            credential_id,
+                            public_key,
+                            cred_protect,
+                            large_blob_key: large_blob_key.map(|x| x.into_vec()),
```

Comment:
I think this is `ByteBuf::into_vec` and we should be careful that this doesn't become an unnecessary copy when we change `ByteBuf` to `Vec` in the struct definition.

Reply:
With the change from `ByteBuf` to `Vec` in the structs, the only place where the conversion from `ByteBuf` to `Vec` happens (using `into_vec()`, which is doesn't copy), is in the deserialize-routine. And there, I added a comment. `CredentialListEntry`, which contains `large_blob_key` is not "Clone" at the moment. So, I think we are safe-ish for the moment (`entry.large_blob_key.clone()` would still work, though. But I can't think of a way to prevent that).